src/parser/LiteralType.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023, University of Freiburg,
+//                 Chair of Algorithms and Data Structures.
+// Author: Benedikt Maria Beckermann <benedikt.beckermann@dagstuhl.de>
+
+#include "LiteralType.h"
+
+#include <utility>
+
+LiteralType::LiteralType(NormalizedString content) {
+  this->content = std::move(content);
+  this->descriptorType = LiteralDescriptor::NONE;
+}
+
+LiteralType::LiteralType(NormalizedString content,
+                         NormalizedString datatypeOrLanguageTag,
+                         LiteralDescriptor type) {
+  this->content = std::move(content);
+  this->descriptorType = type;
+  this->descriptorValue = std::move(datatypeOrLanguageTag);
+}
+
+bool LiteralType::hasLanguageTag() const {
+  return this->descriptorType == LiteralDescriptor::LANGUAGE_TAG;
+}
+
+bool LiteralType::hasDatatype() const {
+  return this->descriptorType == LiteralDescriptor::DATATYPE;
+}
+
```

Comment:
No `this`, and add empty comments above all functions
```
// __________________________________________
bool blad() {...}
```

(Our convention to make the beginning and end of functions more visible).